File: LivrusVirtus/BoaConstructor/wxPython/CadEditoras.py
#Boa:FramePanel:frmCadEditoras

from wxPython.wx import *
from TextValidator import TextValidator
from Cadastros import frmCadastros
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource

def create(parent):
    return frmCadEditoras(parent)

class frmCadEditoras(frmCadastros):
    def _init_utils(self):
        pass

    def _init_ctrls(self, prnt):
        frmCadastros._init_ctrls(self, prnt, TextValidator.FILTER_NUMERIC, True)
        self._init_utils()
        self.labDescricao.SetLabel(u'Editora:')
        EVT_BUTTON(self.btnNovo, self.btnNovo.GetId(), self.OnBtnNovoClick)
        EVT_BUTTON(self.btnFechar, self.btnFechar.GetId(), 
          self.OnBtnFecharClick)
        EVT_BUTTON(self.btnPesquisar, self.btnPesquisar.GetId(),
              self.OnBtnPesquisarClick)
        EVT_BUTTON(self.btnAnterior, self.btnAnterior.GetId(),
              self.OnBtnAnteriorClick)
        EVT_BUTTON(self.btnProximo, self.btnProximo.GetId(),
              self.OnBtnProximoClick)
        EVT_BUTTON(self.btnSalvar, self.btnSalvar.GetId(),
              self.OnBtnSalvarClick)
        EVT_BUTTON(self.btnExcluir, self.btnExcluir.GetId(),
              self.OnBtnExcluirClick)
        EVT_BUTTON(self.btnLimpar, self.btnLimpar.GetId(),
              self.OnBtnLimparClick)
                
    def __init__(self, parent):
        self._init_ctrls(parent)
        
    def ValidaDados(self, bTodosDados):
      if RotinasGlobais.Rotinas.ValidaCampo(
        self.edtCodigo.GetValue(), Resource.STR_CODNINF):
        return False
      if bTodosDados:
    	  if RotinasGlobais.Rotinas.ValidaCampo(
          self.edtDescricao.GetValue(), Resource.STR_EDTNINF):
      		return False;
      return True

    def OnBtnFecharClick(self, event):
      self.GetParent().Close(True)
    
    def OnBtnNovoClick(self, event):
      self.btnNovoClick()
      self.NovoDado(ConsultasSQL.ConsSQL.Editora('N', 
        self.edtCodigo.GetValue(),""))
      event.Skip()

    def OnBtnPesquisarClick(self, event):
      self.PesquisarDados(ConsultasSQL.ConsSQL.Editora('S', 
        self.edtCodigo.GetValue(),""), Resource.STR_EDTNENC)
      event.Skip()

    def OnBtnAnteriorClick(self, event):
      self.registroAnterior()
      event.Skip()

    def OnBtnProximoClick(self, event):
      self.registroProximo()
      event.Skip()

    def OnBtnSalvarClick(self, event):
      self.SalvarDados(self.ValidaDados(True),
        ConsultasSQL.ConsSQL.Editora('S', self.edtCodigo.GetValue(),""),
        ConsultasSQL.ConsSQL.Editora('U', self.edtCodigo.GetValue(), 
        self.edtDescricao.GetValue()),
        ConsultasSQL.ConsSQL.Editora('I', self.edtCodigo.GetValue(), 
        self.edtDescricao.GetValue()), True)
      event.Skip()

    def OnBtnExcluirClick(self, event):
      self.ExcluirDados(self.ValidaDados(False),
        ConsultasSQL.ConsSQL.Editora('D', self.edtCodigo.GetValue(),""))
      event.Skip()

    def OnBtnLimparClick(self, event):
      self.btnLimparClick()
      event.Skip()

File: LivrusVirtus/VCPP6/FLTK/LibPostgreSQL/CadEditoras.cpp
#include "Principal.h"
#include "CadEditoras.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"

CfrmCadEditoras *frmCadEditoras;

void frmCadEditoras_btnNovoClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnPesquisarClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnAnteriorClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnProximoClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnSalvarClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnExcluirClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnLimparClick_cb(Fl_Button* b, void* v);
void frmCadEditoras_btnFecharClick_cb(Fl_Button* b, void* v);

CfrmCadEditoras::CfrmCadEditoras()
       : CfrmCadastros(true, STR_CADEDITORA) {
	IniciaControles();
}

CfrmCadEditoras::~CfrmCadEditoras() { }

void CfrmCadEditoras::IniciaControles() {
  edtDescricao->label("Editora:");

  btnNovo->callback((Fl_Callback*)frmCadEditoras_btnNovoClick_cb);
  btnPesquisar->callback((Fl_Callback*)frmCadEditoras_btnPesquisarClick_cb);
  btnAnterior->callback((Fl_Callback*)frmCadEditoras_btnAnteriorClick_cb);
  btnProximo->callback((Fl_Callback*)frmCadEditoras_btnProximoClick_cb);
  btnSalvar->callback((Fl_Callback*)frmCadEditoras_btnSalvarClick_cb);
  btnExcluir->callback((Fl_Callback*)frmCadEditoras_btnExcluirClick_cb);
  btnLimpar->callback((Fl_Callback*)frmCadEditoras_btnLimparClick_cb);
  btnFechar->callback((Fl_Callback*)frmCadEditoras_btnFecharClick_cb);
}

bool CfrmCadEditoras::ValidaDados(bool bTodosDados)
{
if (Rotinas->ValidaCampo(edtCodigo->value(), STR_CODNINF))
  	  return false;
  	if (bTodosDados) {
  	  if (Rotinas->ValidaCampo(edtDescricao->value(), STR_EDTNINF))
  		return false;
  	}
  	return true;
}

void frmCadEditoras_btnNovoClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->btnNovoClick();
  frmCadEditoras->NovoDado(ConsSQL->Editora('N', 
    frmCadEditoras->edtCodigo->value(),""));
}

void frmCadEditoras_btnPesquisarClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->PesquisarDados(ConsSQL->Editora('S', 
    frmCadEditoras->edtCodigo->value(),""), STR_EDTNENC);
}

void frmCadEditoras_btnAnteriorClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->registroAnterior();	
}

void frmCadEditoras_btnProximoClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->registroProximo();
}

void frmCadEditoras_btnSalvarClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->SalvarDados(frmCadEditoras->ValidaDados(true),
    ConsSQL->Editora('S', frmCadEditoras->edtCodigo->value(),""),
    ConsSQL->Editora('U', frmCadEditoras->edtCodigo->value(), 
    frmCadEditoras->edtDescricao->value()),
    ConsSQL->Editora('I', frmCadEditoras->edtCodigo->value(), 
    frmCadEditoras->edtDescricao->value()), true);
}

void frmCadEditoras_btnExcluirClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->ExcluirDados(frmCadEditoras->ValidaDados(false),
    ConsSQL->Editora('D', frmCadEditoras->edtCodigo->value(),""));
}

void frmCadEditoras_btnLimparClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->btnLimparClick();
}

void frmCadEditoras_btnFecharClick_cb(Fl_Button* b, void* v){
  frmCadEditoras->hide();
}

int CfrmCadEditoras::handle(int event) {
  if (event == FL_HIDE) {
    frmPrincipal->BarraDeStatus2->label("");
    frmPrincipal->mnuCadEditoras->activate();
  }
  return CfrmCadastros::handle(event);
}
